server/src/main.rs

```diff
@@ -2,6 +2,9 @@ mod collection;
 mod error;
 mod item;
 mod prompt;
+mod prompter;
+mod prompter_callback;
+mod secret_exchange;
```

Comment:
Those two seems to be specific to the gnome prompting mechanism, so probably should go under a gnome sub-module.